### DISCUSSION TITLE
Is there a tool to eliminate a variable from an exodus file?

### DISCUSSION THREAD

**Discussion by bartgol** [trilinos/Trilinos#12556] [Q&A] [answered]
I have an exodus file, and I would like to remove a variable from it. Is there a tool shipped with SEACAS that allows it? I looked at `explore`, but I couldn't really figure out if it's possible.

My first guess was to use nco tools (ncks, specifically) to remove the variable, but then I would have to
 - "shift" all the variableN that came after my var to a lower index
 - resize the `names_nod_var` field, shifting names "left" as needed
 - update the `num_nod_var` dimension
 
and similarly for elem vars. I figured "maybe someone else already wrote a script for this?".
 
 @trilinos/seacas

**Comment by gsjaardema** [comment, marked as answer]
You can use the `algebra` code to do this.  Syntax is:
```
algebra {input_file} {output_file}
save all
delete {variable_to_delete}
exit
```
Some documentation is at https://sandialabs.github.io/seacas-docs/sphinx/html/index.html#algebra

**Comment by bartgol** [comment]
@gsjaardema Is there a way to run this inside a script? That is, can one forward the commands (like `save all`) to the ALG runtime in a non-interactive way? I'm trying to delete a var as part of a testing workflow...

**Comment by jhux2** [comment]
> Is there a way to run this inside a script?

@bartgol When I needed to automatically verify that thousands of parallel mesh files were not corrupt with the SEACAS `explore` tool, I resorted to some `bash` trickery.  Maybe you could adapt this for your case.  Here's what I did:

```bash
#!/bin/bash

for f in *
do
  export flag=`explore $f <<< check |& egrep "Database check is complete" | wc -l`
  if [ "$flag" -eq "1" ]; then
    echo "$f : ok"
  else
    echo "$f : BAD"
  fi

done
```
